src/hotspot/os/linux/cgroupV1Subsystem_linux.cpp

```diff
@@ -241,3 +241,29 @@ int CgroupV1Subsystem::cpu_shares() {
 
   return shares;
 }
+
+
+char* CgroupV1Subsystem::pids_max_val() {
+  GET_CONTAINER_INFO_CPTR(cptr, _pids, "/pids.max",
+                     "Maximum number of tasks is: %s", "%s %*d", pidsmax, 1024);
+  if (pidsmax == NULL) {
+    return NULL;
+  }
+  return os::strdup(pidsmax);
+}
+
+/* pids_max
+ *
+ * Return the maximum number of tasks available to the process
+ *
+ * return:
+ *    maximum number of tasks
+ *    -1 for unlimited
+ *    OSCONTAINER_ERROR for not supported
+ */
+jlong CgroupV1Subsystem::pids_max() {
+  if (_pids == NULL) return OSCONTAINER_ERROR;
+  char * pidsmax_str = pids_max_val();
+  jlong pidsmax = limit_from_str(pidsmax_str);
+  return pidsmax;
```

Comment:
Do we need this local variable? Consider using `return limit_from_str(pidsmax_str);` instead.